### PR TITLE
[G-API] Wrap cv::gapi::mean kernel into python

### PR DESCRIPTION
**Pull request by TolyaTalamanov**
### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [x] I agree to contribute to the project under Apache 2 License.
- [ ] To the best of my knowledge, the proposed patch is not based on a code under GPL or other license that is incompatible with OpenCV
- [x] The PR is proposed to proper branch
- [ ] There is reference to original bug report and related work
- [x] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [ ] The feature is well documented and sample code can be built with the project CMake

### Overview
Basically this PR proves that opencv `python gen` supports overloads

* Wrapped `cv::gapi::mean` operation
* Wrapped `GScalar`
* Wrapped corresponding overloads for `GComputation()` and `apply`

### Buildbot configuration

```
build_gapi_standalone:Linux x64=ade-0.1.1f
build_gapi_standalone:Win64=ade-0.1.1f
build_gapi_standalone:Mac=ade-0.1.1f
build_gapi_standalone:Linux x64 Debug=ade-0.1.1f

force_builders=ARMv7,Custom
build_image:Custom=centos:7
buildworker:Custom=linux-1
build_gapi_standalone:Custom=ade-0.1.1f
```

### CODE REVIEW

File: modules/gapi/misc/python/test/test_gapi_core.py [hunk 1]
```diff
@@ -37,6 +37,23 @@ def test_add(self):
             # Comparison
             self.assertEqual(0.0, cv.norm(expected, actual, cv.NORM_INF))
 
+    def test_mean(self):
+        sz = (1280, 720, 3)
+        in1 = np.random.randint(0, 100, sz).astype(np.uint8)
+
+        # OpenCV
+        expected = np.mean(in1,axis=(0,1))
```
Comment (TolyaTalamanov) [inline modules/gapi/misc/python/test/test_gapi_core.py:45]: BTW, it isn't `opencv` reference it is `numpy`, does it make sense to compare with opencv ?

**Comment by TolyaTalamanov** [comment]
@dmatveev

File: modules/gapi/misc/python/test/test_gapi_core.py (hunk 1 above)
Comment (dkurt) [reply modules/gapi/misc/python/test/test_gapi_core.py:45]: G-API is positioned as optimized way to rewrite OpenCV algorithms so I think it's useful to see alternatives from OpenCV core, right.

File: modules/gapi/misc/python/test/test_gapi_core.py [hunk 2]
```diff
@@ -37,6 +37,23 @@ def test_add(self):
...
+
+        # G-API
+        g_in1 = cv.GMat()
```
Comment (dkurt) [inline modules/gapi/misc/python/test/test_gapi_core.py:48]: `g_in`?

File: modules/gapi/misc/python/test/test_gapi_core.py [hunk 3]
```diff
@@ -37,6 +37,23 @@ def test_add(self):
             # Comparison
             self.assertEqual(0.0, cv.norm(expected, actual, cv.NORM_INF))
 
+    def test_mean(self):
+        sz = (1280, 720, 3)
+        in1 = np.random.randint(0, 100, sz).astype(np.uint8)
```
Comment (dkurt) [inline modules/gapi/misc/python/test/test_gapi_core.py:42]: `in` or `inp`?

File: modules/gapi/misc/python/test/test_gapi_core.py [hunk 4]
```diff
@@ -37,6 +37,23 @@ def test_add(self):
...
+        # OpenCV
+        expected = np.mean(in1,axis=(0,1))
+
+        # G-API
+        g_in1 = cv.GMat()
+        g_out = cv.gapi.mean(g_in1)
+        comp = cv.GComputation(g_in1, g_out)
+
+        for pkg in pkgs:
+            actual = np.array(comp.apply(in1, args=cv.compile_args(pkg))[:3])
```
[the remark this answers is not in the document]
Comment (TolyaTalamanov) [reply modules/gapi/misc/python/test/test_gapi_core.py:53]: Scalar is a `tuple` with 4 values

[the remark this answers is not in the document]
Comment (dkurt) [reply modules/gapi/misc/python/test/test_gapi_core.py:53]: so maybe if we will use OpenCV as a reference we can avoid it, right?

File: modules/gapi/misc/python/test/test_gapi_core.py (hunk 1 above)
Comment (TolyaTalamanov) [reply modules/gapi/misc/python/test/test_gapi_core.py]: Fixed

File: modules/gapi/misc/python/test/test_gapi_core.py (hunk 2 above)
Comment (TolyaTalamanov) [reply modules/gapi/misc/python/test/test_gapi_core.py]: Fixed

File: modules/gapi/misc/python/test/test_gapi_core.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (TolyaTalamanov) [reply modules/gapi/misc/python/test/test_gapi_core.py]: Fixed

**Review by dkurt — APPROVED** [review]
:+1: Well done!

**Comment by TolyaTalamanov** [comment]
@dmatveev  @alalek

**Review by dmatveev — APPROVED** [review]
:+1: thanks.

Still interested in the generic `gin`/`gout` version though.

Note with `gout` we may also write to a pre-allocated buffer. Would it work in case of python?
